06.wc/wc.rb

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require 'date'
+
+def main
+  options, argv_files = parse_argv
+  text_details = argv_files.empty? ? [get_text_detail($stdin.read)] : get_text_details(argv_files)
+  output_lines(text_details, options)
+end
+
+def parse_argv
+  option = OptionParser.new
+  options = {}
+  option.on('-l') { |v| options[:line] = v }
+  option.on('-w') { |v| options[:word] = v }
+  option.on('-c') { |v| options[:byte] = v }
+  argv_files = option.parse!(ARGV)
+  if options.empty?
+    options[:line] = true
+    options[:word] = true
+    options[:byte] = true
+  end
+  [options, argv_files]
+end
+
+def get_text_details(file_names)
+  file_names.map do |file_name|
+    file_sentense = File.read(file_name)
+    get_text_detail(file_sentense, file_name)
+  end
+end
+
+def get_text_detail(sentence, file_name = nil)
+  {
+    line: sentence.lines.count,
+    word: sentence.split(/\s+/).size,
+    byte: sentence.length,
+    name: file_name
+  }
+end
+
+def output_lines(text_details, options)
+  total_value = calc_total(text_details)
+  col_width = total_value.values_at(:line, :word, :byte).max.to_s.length
+  text_details.each do |text_detail|
+    output_line(text_detail, col_width, options)
+  end
+  return if text_details.size == 1
+
+  output_line(total_value, col_width, options)
+end
+
+def calc_total(text_details)
+  total = Hash.new { |h, k| h[k] = 0 }
+  text_details.each do |text_detail|
+    total[:line] += text_detail[:line]
+    total[:word] += text_detail[:word]
+    total[:byte] += text_detail[:byte]
+  end
+  total[:name] = 'total'
+  total
```

Comment:
`each_with_object`も試してみましたが、出力処理をまとめたくてハッシュに`total[:name] = 'total'`を追加したため、'each'メソッドにしました。
（`each_with_object`はrubocopで警告が出てしまったことも使わなかった理由の一つです。Lint/RedundantWithObject: Use each instead of each_with_object.）

Reply:
```ruby
total = Hash.new { |h, k| h[k] = 0 }
total[:name] = 'total'
text_details.each_with_object ...
```

のようにすれば警告を避けられるかも？

Reply:
total[:name]を追加するという感覚から後に付ける意識しかありませんでした。確かに前に付ければ`each_with_object`の利点が活きますね。

ただ、変更しても同様の警告が出てしまいました。
とはいえ、提出物にも記載しましたが、戻り値を別途指定せずに済みコードがスッキリするという利点から、あえて警告を無視しています。

Reply:
これ、よく見たらブロックパラメータの使い方がちょっと間違ってますね。
公式リファレンスのコード例を見直してみましょう。
https://docs.ruby-lang.org/ja/latest/method/Enumerable/i/each_with_object.html

ただ、僕も手元で直そうとしてみたんですが、rubocopの警告を消すためにコードを修正するとよけいに可読性が落ちたので、今回は無理にeach_with_objectを使わない方がいいかもしれません。

なお、rubocopをパスさせるのは必須条件ですし、rubocopの警告に納得がいかない場合はrubocop-fjordにissueを投げてルールを変えてもらう必要があります〜。

Reply:
たしかに、memo_objを指定してませんでした…
一応、以下コードで警告無しで実現できたものの、確かに可読性が落ちるかなと思い`each`メソッドで書くことにしました。

```
  total = { line: 0, word: 0, byte: 0, name: 'total' }
  text_details.each_with_object(total) do |text_detail, total_content|
    total_content[:line] += text_detail[:line]
    total_content[:word] += text_detail[:word]
    total_content[:byte] += text_detail[:byte]
  end
```

また、最初の行（以下のコード）は
```
  total = { line: 0, word: 0, byte: 0, name: 'total' }
```
別のところで指摘のあった
> 出力対象のキーを明示的に書く方がベター

という話を思い出したことと、nameキーの追加も１行にまとめられるため追加で修正しました。

Reply:
この修正、良いと思います！ 👍